File: 70.climbing-stairs.java
// @leet imports start
import java.util.*;
import java.math.*;
// @leet imports end

// @leet start
class Solution {
    public int climbStairs(int n) {
		int previous=2;
		int current=3;
		if(n<=2)return n;
		for (int i = 3; i < n; i++) {
			int next = previous+current;
			previous=current;
			current=next;
		}
		return current;
    }
}
// @leet end

File: 70.climbing-stairs.py
# @leet start
class Solution:
    def climbStairs(self, n: int) -> int:
        previous = 2
        current = 3
        if n <= 2:
            return n
        for i in range(n - 3):
            next = previous + current
            previous = current
            current = next
        return current


# @leet end

